cmd/state-svc/main.go

```diff
@@ -69,6 +70,13 @@ func main() {
 	}
 	rollbar.SetupRollbar(constants.StateServiceRollbarToken)
 	rollbar.SetConfig(cfg)
+	api.SetConfig(cfg)
+
+	if err := api.RegisterConfigListener(cfg); err != nil {
+		multilog.Critical("Could not register config listener: %v", errs.JoinMessage(err))
+		exitCode = 1
+		return
+	}
```

Comment:
You're already calling `api.SetConfig()`, that ought to take care of registering the config listener imo.